app/backend/src/couchers/servicers/communities.py

```diff
@@ -144,6 +145,86 @@ def ListCommunities(self, request, context, session):
             next_page_token=encrypt_page_token(str(offset + page_size)) if len(nodes) > page_size else None,
         )
 
+    def SearchCommunities(self, request, context, session):
+        raw_q = (request.query or "").strip()
+        if not raw_q:
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, "query_is_empty")
+        if len(raw_q) < 3:
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, "query_too_short")
+
+        req_size = request.page_size or MAX_PAGINATION_LENGTH
+        page_size = max(1, min(MAX_PAGINATION_LENGTH, req_size))
+
+        # page_token → offset with validation
+        try:
+            offset = int(decrypt_page_token(request.page_token)) if request.page_token else 0
+            if offset < 0:
+                raise ValueError
+        except Exception:
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, "invalid_page_token")
```

Comment:
Please don't wrap in exception, there's no need to catch the case of an invalid decrypt token. If that happens it's good that it throws an ugly error and sends it to us in sentry -- it shouldn't happen normally, it means either some user is doing something dodgy (good for us to know), or we have a programming error (also really important for us to know).